src/Attribute/EarlyValidation.php

```diff
@@ -0,0 +1,38 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Input\Validation\Attribute;
+
+use Attribute;
+use Yiisoft\Hydrator\ParameterAttributeInterface;
+use Yiisoft\Validator\RuleInterface;
+
+#[Attribute(Attribute::TARGET_PROPERTY | Attribute::TARGET_PARAMETER | Attribute::IS_REPEATABLE)]
+final class EarlyValidation implements ParameterAttributeInterface
```

Comment:
Should it be `early`?
What if I pass `EarlyValidation` as third property attribute, would it be `early` as well?

Reply:
It's validation before value type casting.